src/controllers/account.js

```diff
@@ -23,8 +23,48 @@ export const createAccount = asyncHandler(async (req, res) => {
     account = await account.save();
 
     if (!account) {
-        throw  new ErrorResponse("Could not create account", BAD_REQUEST);
+        throw new ErrorResponse("Could not create account", BAD_REQUEST);
     }
 
     res.status(CREATED).json({ message: "Account created successfully", account });
 });
+
+export const getAccount = asyncHandler(async (req, res) => {
+    console.info("Getting Account details...");
+
+    const account = await checkAccountExists(req.params.id, "_id");
+
+    if (!account) {
+        throw new ErrorResponse("Account not found", NOT_FOUND);
+    }
+
+    res.status(OK).json({ message: "Account retrieved successfully", account });
+});
+
+export const updateAccountWithId = asyncHandler(async (req, res) => {
+    console.info("updating Account details using id...");
+
+    const account = await checkAccountExists(req.params.id, "_id");
+
+    if (!account) {
+        throw new ErrorResponse("Account not found", NOT_FOUND);
+    }
+
+    const data = await Account.findByIdAndUpdate(req.params.id, req.body, { new: true });
```

Comment:
I think you might need to use `$set` here, just to follow best practices.
```suggestion
    const data = await Account.findByIdAndUpdate(req.params.id, $set: { req.body } , { new: true });
```